---
title: "RNA Seq Analysis_iGenome"
author: "Ke Tan"
date: "10/1/2020"
output: 
  md_document:
    variant: markdown_github
---


## Bash, on clusters

Download Mus Musculus genome from iGenome using wget, and then unzip
(reference: https://www.interserver.net/tips/kb/extract-tar-gz-files-using-linux-command-line/)
```{bash eval=FALSE, include=FALSE}
wget http://igenomes.illumina.com.s3-website-us-east-1.amazonaws.com/Mus_musculus/NCBI/GRCm38/Mus_musculus_NCBI_GRCm38.tar.gz
tar xvzf Mus_musculus_NCBI_GRCm38.tar.gz
```

Run STAR genome index generator to create genome indices
Warning: uses high RAM
sjdbOverhang = sequence length - 1
-n 1 -c 5 -m 24g
Warning: module purge at first, so that you can use STAR module

FIXME: look for the newest conda version of STAR
```{bash eval=FALSE, include=FALSE}
module load STAR/2.7.1a-foss-2016b; STAR --runThreadN 5 --runMode genomeGenerate --genomeDir GenomeIndices --genomeFastaFiles genome.fa --sjdbGTFfile genes.gtf --sjdbOverhang 100
```

Align reads
Check the input (zipped or not!) and output!!
Time: ~4min for one sample
Output size: ~2GB
module purge
```{bash eval=FALSE, include=FALSE}
module load STAR/2.7.1a-foss-2016b; for i in {8,9}; do STAR --genomeDir GenomeIndices --runThreadN 20 --readFilesIn SeqData/${i}_R1_output_forward_paired.fq SeqData/${i}_R2_output_reverse_paired.fq --outFileNamePrefix ReadAlignment/Sample${i} --outSAMtype BAM SortedByCoordinate --outSAMattributes Standard; done
```

Count features
Apparently, the gtf file downloaded from iGenome is not compatible with featureCounts, fixed the format
(reference: https://www.biostars.org/p/173233/)

FIXME: module purge, but you need conda here...

Time: ~20 seconds for one sample
You don't need clusters here, but pay attention to threads
s=0 is best here (Sarah's criteria, but why?)
```{bash eval=FALSE, include=FALSE}
perl -ne 'chomp; @a=split/\t/; %h=split(/ /,$a[8]); $a[8]=join(" ",("gene_id",$h{"gene_id"},"transcript_id",$h{"transcript_id"})); print join("\t",@a),"\n";' genes.gtf > genes_fixed.gtf

for j in {0,1,2}; do for i in {8,9}; do featureCounts -p -a genes_fixed.gtf -g gene_id -t exon -o FeatureCounts/FeatureCounts_Sample${i}_S${j}.txt -s ${j} -T 20 ReadAlignment/Sample${i}Aligned.sortedByCoord.out.bam; done; done
```

Clean up featurecount matrix for R plotting
and manually remove the information at the beginning, and change header of count...
there are 448 more lines than Malay's analysis due to transcript_id

```{bash eval=FALSE, include=FALSE}
for i in {8,9}; do cut -f 1,7 FeatureCounts_Sample${i}_S0.txt > CleanFeatureCounts_Sample${i}_S0.txt; done
for i in {8,9}; do grep -v 'transcript_id' CleanFeatureCounts_Sample${i}_S0.txt > CleanFeatureCounts_Sample${i}_S0_fixed.txt; done
```

Comparison, using R

```{r echo=TRUE, out.height=50, out.width=50}
for (i in 8:9){
  KeArray = read.delim(paste0('/Users/tanke/RStudio/CleanFeatureCounts_Sample',i,'_S0_fixed.txt'),header = TRUE, sep = '', dec = '.')
  MalayArray = read.delim(paste0('/Users/tanke/RStudio/CleanCounts_MalayRSEM_MouseAlignment_Sample',i,'.txt'),header = TRUE, sep = '', dec = '.')
  InnerJoin <- dplyr::inner_join(KeArray,MalayArray,by='Geneid')
  KeCount = InnerJoin[,2]
  MalayCount = InnerJoin[,3]
  library("ggpubr")
  print(ggscatter(InnerJoin, x = "count.x", y = "count.y", 
          add = "reg.line", 
          add.params = list(color = "black", fill = "lightgray", linetype = 3),
          conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coef.size = 5,
          xlab = "Analysis1_Ke", ylab = "Analysis2_Malay",
          title = 'RNASeq Analysis Comparison'))
}
```


